core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/http/pipeline/stages/MakeAsyncHttpRequestStage.java

```diff
@@ -119,59 +117,12 @@ public MakeAsyncHttpRequestStage(TransformingAsyncResponseHandler<Response<Outpu
         return toReturn;
     }
 
-    private static final class WrappedErrorForwardingResponseHandler<T>
```

Comment:
I'm not familiar with this code so I had to do some digging as well.

The only thing this transformer does is also propagate the exception from `TransformingAsyncResponseHandler#onError` to the wrapped `responseFuture`.

It seems like this was done because [CombinedResponseAsyncHttpResponseHandler](https://github.com/aws/aws-sdk-java-v2/blob/2f4b44d7a08a4e2f0b26fe26b0325b065e4547c3/core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/http/async/CombinedResponseAsyncHttpResponseHandler.java#L83-L93) (also introduced with this change) always relies on the future returned from `prepare()` to always be completed in order to further delegate to the right handler implementation.

Reply:
To close the loop, discussed offline and agreed this class is not needed.